firebase-firestore/src/main/java/com/google/firebase/firestore/CollectionReference.java

```diff
@@ -56,10 +56,10 @@ public String getId() {
   }
 
   /**
-   * Gets a DocumentReference to the document that contains this collection. Only subcollections are
-   * contained in a document. For root collections, returns null.
+   * Gets a {@link DocumentReference} to the document that contains this collection. Only
+   * subcollections are contained in a document. For root collections, returns null.
    *
-   * @return The DocumentReference that contains this collection or null if this is a root
+   * @return The {@link DocumentReference} that contains this collection or null if this is a root
```

Comment:
This will add considerable visual noise for no actual benefit. I think we should mostly undo this, or in cases where we're concretely talking about the type and it's important mark those as `{@code}`. `{@link}` should be used to only to refer to types or methods that aren't otherwise implied by the types in the signature.

Look at the current [public method summary](https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/CollectionReference#public-method-summary). Also look at the specific doc for getParent (a direct link to which doesn't seem to work). It's already contextually the case that any type mentioned in the signature is linked.

Here's how this [looks in the JDK](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#insert(int,%20java.lang.CharSequence,%20int,%20int)). In this example `CharSequence` is the type of a parameter of the method so it's merely marked as code, not linked.

By contrast, [this method](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#insert(int,%20boolean)) links to `String.valueOf(boolean)` because that link isn't implied by the types of the method.

In JDK javadoc, they frequently go even further and spell words out when referring to the concept behind the type rather than referring to the type name, e.g. [here they spell "character sequence"](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#append(java.lang.CharSequence) rather than `CharSequence`. This doesn't seem to be applied consistently though.

In any case, adding `{@link}` to all non-self types is overkill and distracting.

Adding `{@code}` to everything is also likely to be overkill though I don't mind that as much. Consider changing to just plain words in most cases when the specific type isn't important.